mx.py

```diff
@@ -2222,6 +2222,9 @@ def resolveDeps(self):
         if licenseId:
             self.theLicense = get_license(licenseId, context=self)
 
+    def substVars(self, text):
+        return text.format(**vars(self))
```

Comment:
If `text` contains references to instance variables defined in the current library instance they will get resolved (because `kwargs` expansion of `vars(self)` provides the instance variables as proper arguments for the standard format function).

For example if `text` is: "http://foo.bar.com/slavefiles/{sha1}/{path}" and the instance has the following instance variables: `self.path = "myfile.jar"` and `self.sha1 = ''a1ca304d9b6842bddbc160a45150590b82b762e7` then calling `self.substVars("http://foo.bar.com/slavefiles/{sha1}/{path}")` will return `"http://foo.bar.com/slavefiles/a1ca304d9b6842bddbc160a45150590b82b762e7/myfile.jar"`